config/prometheus/monitor.yaml

```diff
@@ -1,16 +1,20 @@
-
 # Prometheus Monitor Service (Metrics)
 apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   labels:
     control-plane: controller-manager
-  name: controller-manager-metrics-monitor
-  namespace: system
+  name: capsule-monitor
+  namespace: capsule-system
 spec:
   endpoints:
-    - path: /metrics
+    - interval: 15s
+      path: /metrics
       port: metrics
+  jobLabel: app.kubernetes.io/name
+  namespaceSelector:
```

Comment:
Do we need the `namespaceSelector` here although the Service Monitor is living inside the Capsule namespace?

The hard thing is that we're not using Helm and I don't know how to inject the `namespace` value provided by the `config/default/kustomization.yaml` file.

Reply:
`namespaceSelector` should not be related to service monitor location ( maybe you're thinking about `ServiceMonitorNamespaceSelector` );
nevertheless, can be left empty without any problems.

Keep in mind that, in this way, servicemonitor will search for services to scrape (by matchLabels) in every namespace.

Reply:
You're totally right.

I checked out what's the scaffolded ServiceMonitor by `operator-sdk create api` command and it's the following:

```yaml
# Prometheus Monitor Service (Metrics)
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    control-plane: controller-manager
  name: controller-manager-metrics-monitor
  namespace: system
spec:
  endpoints:
    - path: /metrics
      port: https
      scheme: https
      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
      tlsConfig:
        insecureSkipVerify: true
  selector:
    matchLabels:
      control-plane: controller-manager
```

Let's use this non-optimal solution for the ServiceMonitor backed by `kustomize`, we got more control over Helm.